include/deal.II/base/types.h

```diff
@@ -41,6 +41,12 @@ namespace types
    */
   typedef unsigned int subdomain_id;
 
+  /**
+   * The type used for global indices of vertices.
+   */
+  typedef unsigned long long int global_vertex_index;
+#  define DEAL_II_VERTEX_INDEX_MPI_TYPE MPI_UNSIGNED_LONG_LONG
```

Comment:
Or is it that you want to avoid differences between DEAL_II_WITH_64BIT_INDICES=ON/OFF?

(Back to my original thought of re-using global_dof_index: I must admit that I am not a big fan of too many typedefs, so my view might be biased.)